docs/usage.md

```diff
@@ -0,0 +1,65 @@
+# Usage
+
+## Rendering
+
+### Basic HTML rendering
+
+The demo project comes with 2 files `src/html.htl` and `src/html.pre.js` which renders HTML from the requested Markdown file. The `htl` file is the rendering template while the `pre.js` allows to interact with the payload before it gets rendered by the template.
+The These files are named based on the following logic: `<selector>_<extension>.htl` and `<selector>_<extension>.pre.js`. In this case, the `selector` is undefined. They will handle all requests coming to `http://localhost:3000/**/*.html`.
+
+The `htl` file format must respect the HTL spec, check the [htlengine](https://github.com/adobe/htlengine) for more details.
+The `pre.js` file must only export a `pre` function, check the [helix-pipeline](https://github.com/adobe/helix-pipeline) for the method signature.
+
+### Custom selector
+
+You can define the following pair `src/myselector_html.htl` and `src/myselector_pre.js` to handle request like `http://localhost:3000/**/*.myselector.html`. You can create as many selector as you want.
+
+One of the usage of the selectors (in combination with ESI - [Edge Server Includes](https://www.w3.org/TR/esi-lang)) could be to have multiple HTML rendering for your site. You could have 3 `htl` files (together with their `pre.js`):
+
+1. `html.htl`
+1. `homepage_html.htl`
+1. `others_html.htl`
+
+The `html.htl` file could look like this:
+
+```html
+<esi:include src="${payload.dispatch_url}" />
+```
+
+The `htmp.pre.js` would be responsible to determine which of the `homepage_html.htl` or `others_html.htl` needs to be used for a given request. Here is an example where only the `index.md` file is rendered with the `homepage_html.htl` script:
+
+```js
+function pre(payload, action) {
```

Comment:
please rename `payload` to `context`. so it matches the [schema docs](https://github.com/adobe/helix-pipeline/blob/master/docs/content.schema.md).